src/main/kotlin/com/lovelysystems/gradle/LSAws.kt

```diff
@@ -0,0 +1,114 @@
+                val configFile = Paths.get("$userHome/.aws/config")
+                val config = Files.readAllLines(configFile)
+
+                if (!config.contains("[sso-session ${SsoSessionSettings.NAME}]")) {
+                    val sessionConfig = ssoSessionSettings.toAwsConfig()
+                    Files.write(configFile, sessionConfig.toByteArray(), StandardOpenOption.APPEND)
+                    println("Added ${SsoSessionSettings.NAME} session config to $configFile")
+                }
+            }
+        }
+
+        fun runCmd(cmd: String, onError: (AwsError) -> Unit = { error(it.msg) }) {
+            ByteArrayOutputStream().use { errorOut ->
+                val res = project.exec {
+                    errorOutput = errorOut
+                    isIgnoreExitValue = true
+                    commandLine("sh", "-c", cmd)
+                }
+                if (res.exitValue != 0) {
+                    val awsError = AwsError(errorOut.toString().trim())
+                    onError(awsError)
+                }
+            }
+        }
+
+        /**
+         * Task to get AWS SSO credentials for the configured [profile].
+         *
+         * - If the profile is not configured, it will abort and ask to configure the profile first.
+         * - If the profile is configured but the SSO token is expired, it will login to AWS SSO (automatically opens
+         *   browser).
+         */
+        val ssoCredentials by registering {
+            group = AWS_GROUP
+            description = "Get credentials using AWS SSO"
+            dependsOn(ssoSessionSetup)
+
+            doLast {
+                runCmd("[[ \$(aws configure --profile $profile list) && \$? -eq 0 ]]") { error ->
+                    if (error.profileNotFoundError) {
+                        val msg = """
+                        AWS profile '$profile' is not configured:
+                         - Run 'aws configure sso --profile ${profile}' to configure the profile.
+                         - Use `${SsoSessionSettings.NAME}` when prompted for the 'SSO session name', for the rest use the default values.
+                        """.trimIndent()
+                        error(msg)
+                    } else {
+                        error("Failed to check if profile '$profile' is configured. Error: ${error.msg}")
+                    }
+                }
+
+                runCmd("aws sts get-caller-identity --profile $profile") { error ->
+                    if (error.ssoTokenError) runCmd("aws sso login --profile $profile")
+                    else error("Failed to get caller identity for profile '$profile'. Error: ${error.msg}")
+                }
+
+                println("\nAWS SSO credentials for profile '$profile' are now configured and ready to be used.")
+            }
+        }
+
+    }
+
+}
+
+data class SsoSessionSettings(
+    val startUrl: String = "https://lovelysystems.awsapps.com/start",
```

Comment:
Somehow a default `startUrl` like this feels a little weird for open source plugin like this, since browsers will open the lovely AWS login by default (e.g. if they miss to configure the startUrl properly). Personally I would make this a required field when using it in the build.gradle, even if it feels redundant for our projects.

Reply:
take this just as informal input and change if you like. not requirement from my side.